nav2_smac_planner/src/node_2d.cpp

```diff
@@ -154,13 +154,16 @@ bool Node2D::backtracePath(CoordinateVector & path)
 
   NodePtr current_node = this;
 
-  do {
+  while (current_node->parent) {
     path.push_back(
       Node2D::getCoords(current_node->getIndex()));
     current_node = current_node->parent;
-  } while (current_node->parent);
+  }
+
+  // add the start pose
+  path.push_back(Node2D::getCoords(current_node->getIndex()));
 
-  return path.size() > 0;
+  return true;
```

Comment:
Rever to path.size() > 0

Reply:
So, the path size will always be greater than 0 because we check 
` if (!this->parent) {
    return false;
  }`
We will always at least two points. The first one from the while and second from the start.

Reply:
Oh - then > 1?

Reply:
Very good point :smile: 

Apparently my defensive programming is defensive about defense :laughing:

Reply:
:)